action/sealedenvelope.go

```diff
@@ -131,21 +134,77 @@ func (sealed *SealedEnvelope) LoadProto(pbAct *iotextypes.Action) error {
 
 func actionToRLP(action Action) (rlpTransaction, error) {
 	var tx rlpTransaction
+	var err error
+
 	switch act := action.(type) {
 	case *Transfer:
 		tx = (*Transfer)(act)
 	case *Execution:
 		tx = (*Execution)(act)
 	case *CreateStake:
-		// convert to contract execution by filling the addr and calldata
-		// addr is pre-defined special addr for web3 staking
-		// like "000000web3stakecreate" for stake create, "000000web3candregister" for candidate register
-		// calldata is the proto-bytes of native staking action
-		// tx.data = proto.marshal(action.Proto())
-	case *Unstake:
-		// and other staking actions
+		tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+			address.StakingCreateAddrHash,
+			act.amount,
+			act.Proto())
+	// case *DepositToStake:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingAddDepositAddrHash,
+	// 		act.amount,
+	// 		act.Proto())
+	// case *ChangeCandidate:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingChangeCandAddrHash,
+	// 		big.NewInt(0),
+	// 		act.Proto())
+	// case *Restake:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingRestakeAddrHash,
+	// 		big.NewInt(0),
+	// 		act.Proto())
+	// case *TransferStake:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingTransferAddrHash,
+	// 		big.NewInt(0),
+	// 		act.Proto())
+	// case *CandidateRegister:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingRegisterCandAddrHash,
+	// 		act.amount,
+	// 		act.Proto())
+	// case *CandidateUpdate:
+	// 	tx, err = wrapStakingActionIntoExecution(act.AbstractAction,
+	// 		address.StakingUpdateCandAddrHash,
+	// 		big.NewInt(0),
+	// 		act.Proto())
+
+	// other staking actions(WIP)
+	// convert to contract execution by filling the addr and calldata
+	// addr is pre-defined special addr for web3 staking
+	// like "000000web3stakecreate" for stake create, "000000web3candregister" for candidate register
+	// calldata is the proto-bytes of native staking action
+	// tx.data = proto.marshal(action.Proto())
 	default:
 		return nil, errors.Errorf("invalid action type %T not supported", act)
 	}
+	if err != nil {
+		return nil, err
+	}
 	return tx, nil
 }
+
+func wrapStakingActionIntoExecution(ab AbstractAction, toAddr address.Hash160, amount *big.Int, data proto.Message) (rlpTransaction, error) {
+	addr, err := address.FromBytes(toAddr[:])
+	if err != nil {
+		return nil, err
+	}
+	pbAct, err := proto.Marshal(data)
+	if err != nil {
+		return nil, err
+	}
+	return &Execution{
+		AbstractAction: ab,
+		contract:       addr.String(),
+		amount:         amount,
```

Comment:
also there's no need to set amount, since the payload (act.Proto()) contains the amount inside